nipype/pipeline/plugins/multiproc.py

```diff
@@ -100,3 +177,126 @@ def _report_crash(self, node, result=None):
+        # Submit first job on the list
+        for jobid in jobids:
+            logger.debug('Next Job: %d, memory (GB): %d, threads: %d' \
+                         % (jobid, self.procs[jobid]._interface.estimated_memory_gb,
+                            self.procs[jobid]._interface.num_threads))
+
+            if self.procs[jobid]._interface.estimated_memory_gb <= free_memory_gb and \
+               self.procs[jobid]._interface.num_threads <= free_processors:
+                logger.info('Executing: %s ID: %d' %(self.procs[jobid]._id, jobid))
+                executing_now.append(self.procs[jobid])
+
+                if isinstance(self.procs[jobid], MapNode):
+                    try:
+                        num_subnodes = self.procs[jobid].num_subnodes()
+                    except Exception:
+                        self._clean_queue(jobid, graph)
+                        self.proc_pending[jobid] = False
+                        continue
+                    if num_subnodes > 1:
+                        submit = self._submit_mapnode(jobid)
+                        if not submit:
+                            continue
+
+                # change job status in appropriate queues
+                self.proc_done[jobid] = True
+                self.proc_pending[jobid] = True
+
+                free_memory_gb -= self.procs[jobid]._interface.estimated_memory_gb
+                free_processors -= self.procs[jobid]._interface.num_threads
+
+                # Send job to task manager and add to pending tasks
+                if self._status_callback:
+                    self._status_callback(self.procs[jobid], 'start')
+                if str2bool(self.procs[jobid].config['execution']['local_hash_check']):
+                    logger.debug('checking hash locally')
+                    try:
+                        hash_exists, _, _, _ = self.procs[
+                            jobid].hash_exists()
+                        logger.debug('Hash exists %s' % str(hash_exists))
+                        if (hash_exists and (self.procs[jobid].overwrite == False or \
+                                             (self.procs[jobid].overwrite == None and \
+                                              not self.procs[jobid]._interface.always_run))):
+                            self._task_finished_cb(jobid)
+                            self._remove_node_dirs()
+                            continue
+                    except Exception:
+                        self._clean_queue(jobid, graph)
+                        self.proc_pending[jobid] = False
+                        continue
+                logger.debug('Finished checking hash')
+
+                if self.procs[jobid].run_without_submitting:
+                    logger.debug('Running node %s on master thread' \
+                                 % self.procs[jobid])
+                    try:
+                        self.procs[jobid].run()
+                    except Exception:
```

Comment:
Works for me! Again - I'm not trying to picky here. Just want all of us to
avoid the situation when something breaks and we will keep scratching our
heads because the relevant exception was swallowed by one of those
"catch-all" snippets.

On Mon, May 2, 2016 at 8:54 AM, Daniel Clark <notifications@github.com>
wrote:

> In nipype/pipeline/plugins/multiproc.py
> <https://github.com/nipy/nipype/pull/1372#discussion_r61758171>:
>
> > +                                              not self.procs[jobid]._interface.always_run))):
> > +                            self._task_finished_cb(jobid)
> > +                            self._remove_node_dirs()
> > +                            continue
> > +                    except Exception:
> > +                        self._clean_queue(jobid, graph)
> > +                        self.proc_pending[jobid] = False
> > +                        continue
> > +                logger.debug('Finished checking hash')
> > +
> > +                if self.procs[jobid].run_without_submitting:
> > +                    logger.debug('Running node %s on master thread' \
> > +                                 % self.procs[jobid])
> > +                    try:
> > +                        self.procs[jobid].run()
> > +                    except Exception:
>
> @chrisfilo <https://github.com/chrisfilo> - what about something like
> this:
> https://github.com/FCP-INDI/nipype/blob/resource_multiproc/nipype/pipeline/plugins/multiproc.py#L240-246
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/nipy/nipype/pull/1372/files/cf08147051c54588a751ee9968e0263522d5a2b0#r61758171>
>